src/components/organisms/ResetPasswordForm/ResetPasswordFormContainer.tsx

```diff
@@ -0,0 +1,111 @@
+import { yupResolver } from "@hookform/resolvers/yup";
+import { useRouter } from "next/router";
+import { FC, useEffect } from "react";
+import { SubmitHandler, useForm } from "react-hook-form";
+import * as yup from "yup";
+import { getErrors } from "@/helpers/apiHelpers";
+import { useLanguage } from "@/language";
+import { useNotification } from "@/providers";
+import { useResetPasswordMutation } from "@/redux/features/auth";
+import { ResetPasswordPayload } from "@/types";
+import { ResetPasswordFormComponent } from "./ResetPasswordFormComponent";
+
+export type ResetPasswordFormData = Omit<ResetPasswordPayload, "resetToken">;
+
+export const ResetPasswordFormContainer: FC = () => {
+  const { t } = useLanguage();
+  const { showNotification } = useNotification();
+  const { push, query } = useRouter();
+
+  const schema = yup.object({
+    password: yup
+      .string()
+      .min(
+        8,
+        t("errors.minAmountOfCharacters", {
+          fieldName: t("general.password"),
+          number: 8,
+        })
+      )
+      .max(
+        100,
+        t("errors.maxAmountOfCharacters", {
+          fieldName: t("general.password"),
+          number: 100,
+        })
+      )
+      .required(
+        t("errors.requiredField", {
+          fieldName: t("general.password"),
+        })
+      )
+      .label(t("general.password")),
+    passwordConfirm: yup
+      .string()
+      .max(
+        100,
+        t("errors.maxAmountOfCharacters", {
+          fieldName: t("general.passwordConfirm"),
+          number: 100,
+        })
+      )
+      .required(
+        t("errors.requiredField", {
+          fieldName: t("general.passwordConfirm"),
+        })
+      )
+      .oneOf([yup.ref("password")], "Passwords must match")
+      .label(t("general.passwordConfirm")),
+  });
+
+  const { formState, handleSubmit, register, watch, trigger, setError } =
+    useForm<ResetPasswordFormData>({
+      mode: "onTouched",
+      resolver: yupResolver(schema),
+      defaultValues: {
+        password: "",
+        passwordConfirm: "",
+      },
+    });
+
+  const [resetPassword, { data, isSuccess, error, isLoading }] =
+    useResetPasswordMutation();
+
+  const onSubmit: SubmitHandler<ResetPasswordFormData> = (data) => {
+    resetPassword({ ...data, resetToken: String(query.token) });
+  };
+
+  useEffect(() => {
+    if (isSuccess) {
+      push("/");
+      showNotification(`${data?.message}. You can log in now.`, "success");
+    }
+  }, [isSuccess, data?.message]);
```

Comment:
Don't do this in useEffect, handle this logic in `onSubmit` when resetPassword succeedes